compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -324,6 +325,10 @@ void compile(CompilerContext compilerContext) {
     }
 
     void generatePlatformSpecificCode(CompilerContext compilerContext, CompilerBackend compilerBackend) {
+        // adding the source type which will be used to avoid bala projects
+        // when injecting observability imports at desugar phase.
+        CompilerOptions options = CompilerOptions.getInstance(compilerContext);
```

Comment:
This might fail in a parallel build right since we update the same context ? alternatively is there a way in observability desugger to determine if it is the current package.

Reply:
We don't have any Project API related information in ObservabilityDesugar class. Only the bLangPackge is available. Will check more for an alternative solution.

Reply:
Fixed with dab93c9